modules/rte_mis_lottery/src/Controller/DataController.php

```diff
@@ -207,25 +167,28 @@ public function post(Request $request) {
     // Check if the queue already has items.
     $queue = $this->queueFactory->get('student_data_lottery_queue_cron');
     if ($queue->numberOfItems() > 0) {
-      return new ResourceResponse(['error' => 'Lottery already in progress.'], 400);
+      return new JsonResponse(['error' => 'Lottery already in progress.'], 400);
     }
 
     try {
       // Add student data to the queue in batches.
       $batchSize = 100;
-      $chunks = array_chunk($student_data, $batchSize);
+      $chunks = array_chunk($student_data, $batchSize, TRUE);
       foreach ($chunks as $chunk) {
         $queue->createItem($chunk);
```

Comment:
Before adding the items to the queue, please randomize the records.

Reply:
ok.